src/app/feed/feed-create-post/feed-create-post.component.ts

```diff
@@ -234,8 +244,64 @@ export class FeedCreatePostComponent implements OnInit {
   }
 
   _handleFileInput(file: File) {
-    if (!file.type || !file.type.startsWith("image/")) {
-      this.globalVars._alertError("File selected does not have an image file type.");
+    if (!file) {
+      return;
+    }
+
+    if (!file.type || (!file.type.startsWith("image/") && !file.type.startsWith("video/"))) {
+      this.globalVars._alertError("File selected does not have an image or video file type.");
+      return;
+    }
+    if (file.type.startsWith("video/")) {
+      // if (file.size > 200 * (1024 * 1024)) {
+      //   this.globalVars._alertError("File is too large. Please choose a file less than 200MB");
+      //   return;
+      // }
+      let upload: tus.Upload;
+      let mediaId = "";
+      const comp: FeedCreatePostComponent = this;
+      const options = {
+        endpoint: this.backendApi._makeRequestURL(environment.uploadVideoHostname, BackendRoutes.RoutePathUploadVideo),
+        chunkSize: 50 * 1024 * 1024, // Required a minimum chunk size of 5MB, here we use 50MB.
+        // resume: true,
+        // metadata: {
+        //   filename: "test.mp4",
+        //   filetype: "video/mp4",
+        //   defaulttimestamppct: 0.5,
+        //   watermark: "$WATERMARKUID"
+        // },
+        uploadSize: file.size,
+        onError: function (error) {
+          comp.globalVars._alertError(error);
+          upload.abort(true);
+          throw error;
+        },
+        onProgress: function (bytesUploaded, bytesTotal) {
+          comp.videoUploadPercentage = ((bytesUploaded / bytesTotal) * 100).toFixed(2);
+        },
+        onSuccess: function () {
+          comp.postVideoSrc = `https://iframe.videodelivery.net/${mediaId}`;
+          comp.postImageSrc = null;
+          comp.videoUploadPercentage = null;
+          comp.pollForReadyToStream();
+        },
+        onAfterResponse: function (req, res) {
+          return new Promise((resolve) => {
+            let mediaIdHeader = res.getHeader("stream-media-id");
+            if (mediaIdHeader) {
+              mediaId = mediaIdHeader;
+            }
+            resolve(res);
+          });
+        },
+      };
+      if (this.videoStreamInterval != null) {
+        clearInterval(this.videoStreamInterval);
+      }
+      this.postVideoSrc = null;
+      this.readyToStream = false;
+      upload = new tus.Upload(file, options);
+      upload.start();
       return;
     }
     if (file.size > 15 * (1024 * 1024)) {
```

Comment:
nit: I think it would improve readability if we moved all the image processing stuff into its own if statement.  I'm imagining something like:
```
if (image) {}
else if (video) {}
else { error } 
```